nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-0-10-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/KafkaProcessorUtils.java

```diff
@@ -257,6 +288,54 @@
         return results;
     }
 
+
+    static final class KafkaConsumerTopicNameValidator implements Validator {
+
+        final Class<?> classType;
+
+        public KafkaConsumerTopicNameValidator(final Class<?> classType) {
+            this.classType = classType;
+        }
+
+        @Override
+        public ValidationResult validate(final String subject, final String value, final ValidationContext context) {
+            final String topicType = context.getProperty(CONSUMER_TOPIC_TYPE).evaluateAttributeExpressions().getValue();
+            final String topicListing = context.getProperty(CONSUMER_TOPICS).evaluateAttributeExpressions().getValue();
+            boolean isValidTopicName = true;
+            String topicName = "";
+            if (topicListing != null && topicType.equals(TOPIC_NAME.getValue())) {
+                for (final String topic : topicListing.split(",", 100)) {
+                    if(topic.isEmpty() || TOPIC_NAME_PATTERN.matcher(topic).replaceAll("").length() != 0 ) {
+                        isValidTopicName = false;
+                        topicName = topic;
+                    }
+                }
+            }
+            return new ValidationResult.Builder().subject(subject).explanation(topicName + " is not a valid topic name.").valid(isValidTopicName).build();
+        }
+    };
+
+    static final class KafkaProducerTopicNameValidator implements Validator {
+
+        final Class<?> classType;
+
+        public KafkaProducerTopicNameValidator(final Class<?> classType) {
+            this.classType = classType;
+        }
+
+        @Override
+        public ValidationResult validate(final String subject, final String value, final ValidationContext context) {
+            final String topicName = context.getProperty(PRODUCER_TOPIC).evaluateAttributeExpressions().getValue();
```

Comment:
I'm not sure if we can validate the producer topic like this due to the fact that it supports EL from flow files. When EL from flow files is supported, flow files attributes are checked before variable registry. So there could be a variable name "kafka.topic" that is invalid, but at runtime the flow files have an attribute named kafka.topic that is valid, but you can't know that until the processor is running.

Reply:
Does it make sense to change to VARIABLE_REGISTRY instead of a flowfile attribute or write a custom validator like here  https://github.com/apache/nifi/blob/4515b2afe9cef93f823678b4a7ca02a07461dbef/nifi-nar-bundles/nifi-kafka-bundle/nifi-kafka-0-10-processors/src/main/java/org/apache/nifi/processors/kafka/pubsub/PublishKafka_0_10.java#L251 ?

or validate in the onTrigger method?

Reply:
We can't change to variable registry because then you can't route different flow files to different topics using a single PublishKafka processor, plus it would be a behavioral change to existing flows that are relying on attributes, so even if we wanted to change it, it could only be done on a major release (2.x).

The only real option is to handle it in onTrigger. Typically you wouldn't really do validation at that point, you would just check that after calling context.getProperty(topic).evaluateAttributeExpressions(flowFile).getValue() you have a non-blank value. If so then you try to publish to kafka, if an exception happens route to failure.

Reply:
@bbende thanks for the review, Will update PR with validation inside the onTrigger method.